cpp/src/arrow/flight/flight_sql/server.h

```diff
@@ -20,24 +20,15 @@
 
 #pragma once
 
-#include <boost/variant.hpp>
 #include <memory>
 #include <string>
 #include <unordered_map>
 
-#include "arrow/flight/flight_sql/example/sqlite_statement.h"
-#include "arrow/flight/flight_sql/example/sqlite_statement_batch_reader.h"
 #include "arrow/flight/flight_sql/server.h"
-#include "arrow/flight/flight_sql/sql_info_util.h"
+#include "arrow/flight/flight_sql/sql_info_internal.h"
```

Comment:
Note that _internal headers will not get installed, so they can't be part of public headers. If we need the typedefs, we should put them in a common header.

Reply:
Fixed by adding a `sql_info_types.h`